apps/etl/extraction/sources/base/handler.py

```diff
@@ -20,38 +21,94 @@ class ValidationResponse(typing.TypedDict):
     status: str
 
 
-class BaseExtraction:
+class BaseExtraction(abc.ABC):
     """
     Handles data extraction.
     """
 
-    @classmethod
+    @abc.abstractmethod
+    def extract(extraction_object: ExtractionData) -> int:
+        """
+        Not Implemented
+        return extraction id
+        """
+        raise NotImplementedError()
+
+    def get_file_extension(self, source=None):
+        source_file_extension_map = {
+            ExtractionData.Source.DESINVENTAR: ".zip",
+            ExtractionData.Source.IBTRACS: ".csv",
+        }
+        return source_file_extension_map.get(source, ".json")
```

Comment:
This should not be defined on base extraction

Reply:
Done